spec/lib/miq_expression/query_helper_spec.rb

```diff
@@ -0,0 +1,142 @@
+RSpec.describe MiqExpression::QueryHelper do
+  describe ".remove_equality" do
+    let(:vm_host_id) { Vm.arel_table[:host_id] }
+    let(:host_id) { Host.arel_table[:id] }
+    let(:host_type) { Host.arel_table[:type] }
+    let(:bind_param) { Arel::Nodes::BindParam.new }
+
+    # query:  nil
+    # result: nil
+    it "handles an empty query" do
+      query = nil
+      result = described_class.remove_equality(query, vm_host_id.eq(host_id))
+      expect(result).to be_nil
+    end
+
+    # query:  "vms"."host_id" = "hosts"."id"
+    # result: nil
+    it "removes a matching query" do
+      query = vm_host_id.eq(host_id)
+      result = described_class.remove_equality(query, vm_host_id.eq(host_id))
+      expect(result).to be_nil
+    end
+
+    # query:  "hosts"."id" = "vms"."host_id"
+    # result: nil
+    it "removes a matching query (opposite order)" do
+      query = host_id.eq(vm_host_id)
+      result = described_class.remove_equality(query, vm_host_id.eq(host_id))
+      expect(result).to be_nil
+    end
+
+    # query:  "hosts"."type" = 'abc'
+    # result: "hosts"."type" = 'abc'
+    it "keeps a non matching query" do
```

Comment:
"non matching query" reads oddly here, and doesn't really help clarify what `.remove_equality` does.

Maybe moving `vm_host_id.eq(host_id)` to a `let(:matching_query)` statement (since it is used in every example) would help clarify what "matching query" is supposed to mean.